examples/o1/Using_chained_calls_for_o1_structured_outputs.ipynb

```diff
@@ -0,0 +1,256 @@
+      "        }\n",
+      "    ]\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "import requests\n",
+    "from openai import OpenAI\n",
+    "\n",
+    "client = OpenAI()\n",
+    "\n",
+    "def fetch_html(url):\n",
+    "    response = requests.get(url)\n",
+    "    if response.status_code == 200:\n",
+    "        return response.text\n",
+    "    else:\n",
+    "        return None\n",
+    "\n",
+    "url = \"https://en.wikipedia.org/wiki/List_of_largest_companies_in_the_United_States_by_revenue\"\n",
+    "html_content = fetch_html(url)\n",
+    "\n",
+    "json_format = \"\"\"\n",
+    "{\n",
+    "    companies: [\n",
+    "        {\n",
+    "            \\\"company_name\\\": \\\"OpenAI\\\",\n",
+    "            \\\"page_link\\\": \\\"https://en.wikipedia.org/wiki/OpenAI\\\",\n",
+    "            \\\"reason\\\": \\\"OpenAI would benefit because they are an AI company...\\\"\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\"\"\"\n",
+    "\n",
+    "o1_response = client.chat.completions.create(\n",
+    "    model=\"o1-preview\",\n",
+    "    messages=[\n",
+    "        {\n",
+    "            \"role\": \"user\", \n",
+    "            \"content\": f\"\"\"\n",
+    "You are a business analyst designed to understand how AI technology could be used across large corporations.\n",
+    "\n",
+    "- Read the following html and return which companies would benefit from using AI technology: {html_content}.\n",
+    "- Rank these propects by opportunity by comparing them and show me the top 3. Return only as a JSON with the following format: {json_format}\"\n",
+    "\"\"\"\n",
+    "        }\n",
+    "    ]\n",
+    ")\n",
+    "\n",
+    "print(o1_response.choices[0].message.content)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the response is already quite good - it returns the JSON with the appropriate responses. However, it runs into the same pitfalls as existing use-cases of prompt-only JSON inference: \n",
+    "- You must manually process this JSON into your type-safe structure\n",
+    "- Model refusals are not [explicitly returned from the API as a separate structure](https://platform.openai.com/docs/guides/structured-outputs/refusals)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Structured Outputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's now do this with [structured outputs.](https://platform.openai.com/docs/guides/structured-outputs). To enable this functionality, we'll chain our `o1-preview` response, with a `gpt-4o-mini` response that can effectively handle the data returned from the original `o1-preview` response. "
```

Comment:
Minor nit, I would reword this sentence a bit. It reads a bit weird to go: "our `o1-preview` response, with a `gpt-4o-mini` response that can effectively handle the data returned from the original `o1-preview` response."

Should be o1-preview response --> gpt-4o-mini request.